src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ProvideObsoleteAttributeMessage.cs

```diff
@@ -24,15 +27,51 @@ public abstract class ProvideObsoleteAttributeMessageAnalyzer : DiagnosticAnalyz
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182166.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(compilationContext =>
+            {
+                var obsoleteAttributeType = compilationContext.Compilation.GetTypeByMetadataName("System.ObsoleteAttribute");
+                if (obsoleteAttributeType == null)
+                {
+                    return;
+                }
+
+                compilationContext.RegisterSymbolAction(sc => AnalyzeSymbol(sc, obsoleteAttributeType),
+                    SymbolKind.NamedType,
+                    SymbolKind.Method,
+                    SymbolKind.Field,
+                    SymbolKind.Property,
+                    SymbolKind.Event);
+            });
+        }
+
+        private void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol obsoleteAttributeType)
+        {
+            var attributes = context.Symbol.GetAttributes();
+            foreach (var attribute in attributes)
+            {
+                if (attribute.AttributeClass.Equals(obsoleteAttributeType))
+                {
+                    // ObsoleteAttribute has a constructor that takes no params and two 
+                    // other constructors that take a message as the first param.
+                    // If there are no arguments specificed or if the message argument is empty
+                    // then report a diagnostic.
+                    if (attribute.ConstructorArguments.IsEmpty || 
+                        string.IsNullOrEmpty(attribute.ConstructorArguments.First().Value as string))
+                    {
+                        var node = attribute.ApplicationSyntaxReference.GetSyntax();
+                        context.ReportDiagnostic(node.CreateDiagnostic(Rule, context.Symbol.Name));
```

Comment:
This is an example of what I was talking about: that if the symbol with the `Obsolete` attribute is in a nested class, I'll bet this won't give a message in exactly the same format as FxCop. Once my PR for CA1812 is in, we should change this to use the extension method `context.Symbol.FormatMemberName()`.